### ISSUE TITLE
zune-jpeg: assertion failure with malformed image

### ISSUE THREAD

**Issue by SludgePhD** [etemesi254/zune-image#162]
This image causes a panic in zune-jpeg:

![httpcam](https://github.com/etemesi254/zune-image/assets/96552222/bd7f8e62-18d9-41c3-add8-833cbd796b82)

```
thread '<unnamed>' panicked at /home/sludge/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zune-jpeg-0.4.11/src/mcu.rs:209:9:
assertion `left == right` failed
  left: 32409600
 right: 33177600
```

While the image *is* clearly malformed (truncated?), it would be nice if it could handle this case nicer (with an error, or a partially decoded image), since libjpeg also accepts it. jpeg-decoder returns a "failed to fill whole buffer" error.

**Comment by etemesi254** [comment]
Slight mistake on my side.

It was an assertion to ensure all pixels were decoded but the image is partially full that was fixed in 1f943d81b1eb189535abfd2afac8f017c57164f3

It seems I forgot to  publish with the new changes.

I've been working on an overhaul for the way we do I/O , now we have support for directly reading from files (it's not the fastest, but you don't need to buffer the whole file in memory now), which now makes me pushing it a breaking change.

I can push a 0.5... version to fix this, but not a 0.4

**Comment by SludgePhD** [comment]
Ah, cool! No worries, a fix in 0.5 is more than fine.

**Comment by etemesi254** [comment]
version `0.5.0-rc0` contains a fix, sorry to keep you waiting
